odoo/models.py

```diff
@@ -3047,10 +3047,10 @@ def _condition_to_sql(self, alias: str, fname: str, operator: str, value, query:
             sql_value = self.env.registry.unaccent(sql_value)
 
         if need_wildcard and not value:
-            return SQL("%s IS NULL", sql_field) if operator in expression.NEGATIVE_TERM_OPERATORS else SQL("TRUE")
+            return SQL("FALSE") if operator in expression.NEGATIVE_TERM_OPERATORS else SQL("TRUE")
 
         sql = SQL("(%s %s %s)", sql_left, sql_operator, sql_value)
-        if value and operator in expression.NEGATIVE_TERM_OPERATORS:
+        if (value and operator in expression.NEGATIVE_TERM_OPERATORS) or (not value and operator == '=' and isinstance(value, str)):
```

Comment:
It will be nice to cover that case in the test with a SQL assertion to show the result wanted.

(also change the condition to make it more explicit: `... or (operator == '=' and value == '')` (same comment for the other similar change)

Reply:
I did that in the beginning and ruff (lint) was not happy.